packages/core/contracts/trader/uniswap-broker/UniswapBroker.sol

```diff
@@ -0,0 +1,171 @@
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Pair.sol";
+import "@uniswap/lib/contracts/libraries/Babylonian.sol";
+import "@uniswap/lib/contracts/libraries/TransferHelper.sol";
+import "@uniswap/lib/contracts/libraries/FullMath.sol";
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router01.sol";
+
+/**
+ * @title UniswapBroker
+ * @notice Trading contract used to arb uniswap pairs to a desired "true" price. Intended use is to arb UMA perpetual
+ * synthetics that trade off peg. This implementation can ber used in conjunction with a DSProxy contract to atomically
+ * swap and move a uniswap market.
+ *
+ */
+
+contract UniswapBroker {
+    using SafeMath for uint256;
+
+ /**
+     * @notice Swaps an amount of either token such that the trade results in the uniswap pair's price being as close as
+     * possible to the truePrice. 
+     * @dev True price is expressed in the ratio of token A to token B.
+     * @dev The caller must approve this contract to spend whichever token is intended to be swapped.
+     * @param uniswapRouter address of the uniswap router used to facilate trades.
+     * @param uniswapFactory address of the uniswap factory used to fetch current pair reserves.
+     * @param tokenA address of the first token in the uniswap pair.
+     * @param tokenA address of the second token in the uniswap pair.
+     * @param truePriceTokenA the nominator of the true price.
+     * @param truePriceTokenB the denominatornominator of the true price.
+     * @param maxSpendTokenA maximum to spend in tokenA. Note can be set to zero, thereby limiting the direction of trade.
+     * @param maxSpendTokenA maximum to spend in tokenB. Note can be set to zero, thereby limiting the direction of trade.
+     * @param to recipient of the trade proceeds.
+     * @param to deadline to limit when the trade can execute. If the tx is mined after this timestamp then revert.
+     */
+
+    function swapToPrice(
```

Comment:
Is this method identical to https://github.com/Uniswap/uniswap-v2-periphery/blob/master/contracts/examples/ExampleSwapToPrice.sol or are there certain differences we should review independently?

Reply:
it's identical except I have changed the `UniswapV2LiquidityMathLibrary` method. See the comment below.